lib/src/operations/cardano_derive_native_script_hash_operation.dart

```diff
@@ -50,15 +50,13 @@ class CardanoDeriveNativeScriptHashOperation extends ComplexLedgerOperation<Stri
 
     await send(sendOperation);
 
-    final void Function() invoker = switch (script) {
-      ParsedNativeScript_Complex() => () async {
-          for (final subscript in script.script.scripts) {
-            await _deriveNativeScriptHashAddScript(send, subscript);
-          }
-        },
-      ParsedNativeScript_Simple() => () => (),
-    };
-    invoker();
```

Comment:
You needed `final FutureOr<void> Function()` given it's async and then use `await invoker()`. That's why it was not working as expected. It was running the whole invoker as an async operation, so it was not waiting for it to finish before going to next line.

Keep it as `if` statement though. This "pattern" of using switch statements returning invokers seems to be error-prone because of this issue (which I did not think of until now).